Add documentation about critical proposal types and MintSnsTokens (#2333)
dfinity/portal: docs/developer-docs/integrations/sns/managing/making-proposals.md

* [`TransferSnsTreasuryFunds`](#transfersnstreasuryfunds).
* [`UpgradeSnsControlledCanister`](#upgradesnscontrolledcanister).
* [`ManageSnsMetadata`](#managesnsmetadata).
* [`MintSnsTokens`](#mintsnstokens).

All of the proposals used to manage an SNS are executed on the SNS governance canister so it helps to have for reference, what the [interface for the governance canister](https://sourcegraph.com/github.com/dfinity/ic@4732d8281404c0a7c1e0a91937ffd0e54f2beced/-/blob/rs/sns/governance/canister/governance.did) is.

> 

Reviewer: @anchpop 
1. does the candid example below also have to be updated?
2. I would probably put the description of the critical proposals here.

Author: 1. You're right, it does
2. let me know if I moved it to the right place!

See the types in the code [here](https://sourcegraph.com/github.com/dfinity/ic@4732d8281404c0a7c1e0a91937ffd0e54f2beced/-/blob/rs/sns/governance/proto/ic_sns_governance/pb/v1/governance.proto?L405) - they are called “action” in the code.
:::

### Critical proposal types

> Some proposal types are considered "critical". These are DeregisterDappCanisters, TransferSnsTreasuryFunds, and MintSnsTokens. Critical proposal types have a bit more strict rules to ensure they are only passed with broad community consensus.

Reviewer 2: ```suggestion
Some proposal types are considered "critical". These are DeregisterDappCanisters, TransferSnsTreasuryFunds, and MintSnsTokens. Critical proposal types have more strict rules to ensure they are only passed with broad community consensus.
```

In the following we list all differences to non-critical proposals.

> 1. Voting Thresholds

Reviewer 2: ```suggestion
1. Voting thresholds
```

   Non-critical proposals types can be passed if 3% of the total voting power votes yes and 50% of the exercised voting power votes yes.

   Critical proposals types can only be passed if 20% of the total voting power votes yes and 67% of the exercised voting power votes yes. 

2. Catch-all following

>    Voters can follow other neurons on critical proposal types, but each neuron has to make an active decision of following for each critical proposals type as the “All topics”-following is not applied to critical proposal types. Users who have multiple neurons can actively vote with just one of them and follow this one neuron with all their other neurons

Reviewer: ```suggestion
   Voters can follow other neurons on critical proposal types, but each neuron has to make an active decision of following for each critical proposals type as the catch-all following “All topics” is not applied to critical proposal types. Users who have multiple neurons can actively vote with just one of them and follow this one neuron with all their other neurons.
```

>    Voters can follow other neurons on critical proposal types, but each neuron has to make an active decision of following for each critical proposals type as the catch-all following “All topics” is not applied to critical proposal types. Users who have multiple neurons can actively vote with just one of them and follow this one neuron with all their other neurons.

Reviewer 2: ```suggestion
   Voters can follow other neurons on critical proposal types, but each neuron has to make an active decision of following for each critical proposals type, as the catch-all following “All topics” is not applied to critical proposal types. Users who have multiple neurons can actively vote with just one of them and follow this one neuron with all their other neurons.
```

3. Voting period
   
   The voting period for critical proposal types is 5-10 days and cannot be changed by the SNS. In contrast, for non-critical proposals the default is 4-8 days and this can be adjusted by each SNS DAO.

>    Critical proposal have a longer voting period to as they require a larger voting participation and it is therefore beneficial to give voters a bit more time to participate.

Reviewer: ```suggestion
   Critical proposal have a longer voting period as they require a larger voting participation and it is therefore beneficial to give voters a bit more time to participate.
```

    description : "UPDATED Sample SNS used for educational purposes";
```

### `MintSnsTokens`

> Each SNS can have SNS tokens in its treasury, but it also has the ability to mint new SNS tokens to a particular user. This is a ["critical" proposal type](#criticalproposals), which means it is subject to higher voting thresholds before it is accepted.

Reviewer: I would copy the "this is a critical proposal" part to the other proposals for which this is true (treasury and deregister canister)

Author: Of course, done!

```

> ## Critical proposals

Reviewer: I would move this on top of "generic proposals" and make it a ### title. I think it can be a sub-section of native proposals (I made another comment close to where I mean so it is hopefully more clear)

> Some proposal types are considered "critical". These are DeregisterDappCanisters, TransferSnsTreasuryFunds, and MintSnsTokens. Critical proposal types are handled differently to ensure they are only passed with broad community consensus.

Reviewer: ```suggestion
Some proposal types are considered "critical". These are DeregisterDappCanisters, TransferSnsTreasuryFunds, and MintSnsTokens. Critical proposal types have a bit more strict rules to ensure they are only passed with broad community consensus.
In the following we list all differences to non-critical proposals.
```

1. Voting Thresholds

   Non-critical proposals types can be passed if 3% of the total voting power votes yes and 50% of the exercised voting power votes yes.

   Critical proposals types can only be passed if 20% of the total voting power votes yes and 67% of the exercised voting power votes yes. 

> 2. Catchall following

Reviewer: ```suggestion
2. Catch-all following
```

>    Users must set following explicitly for critical proposal types.

Reviewer: I would add a setence here what catch-all following means. 
Maybe we can just copy some of the explanation from the forum post, e.g., a part of "Remove critical proposals from the catch-all “All topics”: Voters can follow other neurons on critical proposals, but each neuron has to make an active decision of following for each critical proposals type as the “All topics”-following is not applied to critical proposals. Users who have multiple neurons can actively vote with just one of them and follow this one neuron with all their other neurons.
It would be possible to reset following on critical proposals in all existing SNSs as an addition to this measure.
Details: This is an alternative to “Remove following for critical proposals” as proposed above that was proposed by the community. This means that the complete removal of following would not be pursued anymore.
Next steps: If the community does not express concerns here, we plan to remove critical proposals from the catch-all and propose it to the NNS as part of the normal SNS release process. We would then discuss in a separate forum thread whether, in addition, following on critical proposals should also be reset in existing SNSs. This addition can then be done separately if this is what the community agrees on." ?

Author: Great point!

> 

Reviewer: One point is missing here: the vote period for critical proposals is also fixed an usually longer than for normal proposals.

Author: Ah, I knew I was forgetting something!
